### PR TITLE
the mop can only absorb liquid within 1 tile

### PR DESCRIPTION
**Pull request by AnywayFarus**
## About The Pull Request

It will no longer be possible to absorb liquid with a mop at a distance of more than one tile

## How This Contributes To The Skyrat Roleplay Experience

No magic w/o bluespace.

## Changelog

:cl: Vishenka0704
fix: mop can no longer suck liquid further than one tile
/:cl:

### CODE REVIEW

File: modular_skyrat/master_files/code/modules/clothing/clothing_variation_overrides/suit.dm [hunk 1]
```diff
@@ -25,7 +25,7 @@
 /obj/item/clothing/suit/jacket
 	supports_variations_flags = CLOTHING_DIGITIGRADE_VARIATION_NO_NEW_ICON
 
-/obj/item/clothing/suit/ianshirt
+/obj/item/clothing/suit/costume/ianshirt
```
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/modules/clothing/clothing_variation_overrides/suit.dm]: :suspect:

File: modular_skyrat/modules/loadouts/loadout_items/loadout_datum_suit.dm [hunk 2]
```diff
@@ -257,7 +257,7 @@ GLOBAL_LIST_INIT(loadout_exosuits, generate_loadout_items(/datum/loadout_item/su
 
 /datum/loadout_item/suit/ianshirt
 	name = "Ian Shirt"
-	item_path = /obj/item/clothing/suit/ianshirt
+	item_path = /obj/item/clothing/suit/costume/ianshirt
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/loadouts/loadout_items/loadout_datum_suit.dm]: :suspect:

File: code/game/objects/items/mop.dm [hunk 3]
```diff
@@ -29,6 +29,8 @@
 	AddElement(/datum/element/liquids_interaction, on_interaction_callback = /obj/item/mop/.proc/attack_on_liquids_turf)
 
 /obj/item/mop/proc/attack_on_liquids_turf(obj/item/mop/the_mop, turf/T, mob/user, obj/effect/abstract/liquid_turf/liquids)
+	if(!in_range(user, T))
+		return
 	var/free_space = the_mop.reagents.maximum_volume - the_mop.reagents.total_volume
 	if(free_space <= 0)
 		to_chat(user, "<span class='warning'>Your mop can't absorb any more!</span>")
```
Comment (GoldenAlpharex) [inline code/game/objects/items/mop.dm]: We should really move that all to a modular file, if you don't know how, I can teach you tomorrow, I'm just too tired tonight.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: code/game/objects/items/mop.dm (hunk 3 above)
Comment (AnywayFarus) [reply code/game/objects/items/mop.dm]: well I will wait

File: modular_skyrat/modules/liquids/code/mop.dm [hunk 4]
```diff
@@ -0,0 +1,13 @@
+/obj/item/mop/proc/attack_on_liquids_turf(obj/item/mop/the_mop, turf/tile, mob/user, obj/effect/abstract/liquid_turf/liquids)
+	if(!in_range(user, tile))
+		return
+	var/free_space = the_mop.reagents.maximum_volume - the_mop.reagents.total_volume
+	if(free_space <= 0)
+		to_chat(user, "<span class='warning'>Your mop can't absorb any more!</span>")
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/liquids/code/mop.dm]:
```suggestion
		to_chat(user, span_warning("Your mop can't absorb any more!"))
```

File: modular_skyrat/modules/liquids/code/mop.dm [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+		return TRUE
+	var/datum/reagents/tempr = liquids.take_reagents_flat(free_space)
+	tempr.trans_to(the_mop.reagents, tempr.total_volume)
+	to_chat(user, "<span class='notice'>You soak the mop with some liquids.</span>")
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/liquids/code/mop.dm]:
```suggestion
	to_chat(user, span_notice("You soak the mop with some liquids."))
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
Just tiny code improvements while you're there.

File: modular_skyrat/modules/liquids/code/mop.dm [hunk 6]
```diff
@@ -0,0 +1,13 @@
+/obj/item/mop/proc/attack_on_liquids_turf(obj/item/mop/the_mop, turf/tile, mob/user, obj/effect/abstract/liquid_turf/liquids)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/liquids/code/mop.dm:13]: If you could do some documentation for this proc, just to explain what is does and what the arguments are there for, it'd be great, and the PR would be good to merge.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/liquids/code/mop.dm [hunk 7]
```diff
@@ -0,0 +1,25 @@
+/**
+ * The procedure for remove water from turf
+ *
+ * The object is called from liquid_interaction element.
+ * The procedure check range of mop owner and tile, then check reagents in mop, if reagents volume < mop capacity - liquids absorbs from tile
+ * In another way, input a chat about mop capacity
+ * Arguments:
+ * * the_mop - Mop what used to absorb liquids(???)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/liquids/code/mop.dm:8]: I'm still not entirely sure why that needs to be an argument, would it be possible to just not have it anymore? I'm not entirely familiar with elements.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/liquids/code/mop.dm (hunk 7 above)
Comment (AnywayFarus) [reply modular_skyrat/modules/liquids/code/mop.dm:8]: it's need to element, I tried to remove it, but the mop did not suck water anymore.

**Review by GoldenAlpharex — APPROVED** [review]

**Pull request merged**